lib/billing/request.ts

```diff
@@ -53,10 +54,17 @@ export function request(config: ConfigInterface) {
 
     const billingConfig = config.billing[plan];
 
-    const returnUrl = buildEmbeddedAppUrl(config)(
-      hashString(`${session.shop}/admin`, HashFormat.Base64),
+    const cleanShopName = session.shop.replace('.myshopify.com', '');
+    const embeddedAppUrl = buildEmbeddedAppUrl(config)(
+      hashString(`admin.shopify.com/store/${cleanShopName}`, HashFormat.Base64),
     );
 
+    const appUrl = `${config.hostScheme}://${config.hostName}`;
+
+    // if provided a return URL, use it, otherwise use the embeded app URL or hosted app URL
+    const returnedUrl =
```

Comment:
Not crazy about this variable name ... `returnedUrl` tends to imply that this was returned from somewhere, where in fact it's the setting to be passed in to the graphql ... maybe `returnUrlForMutation`?